pkg/sdkstats/sdkstats.go

```diff
@@ -2,161 +2,120 @@ package sdkstats
+	Name         string
+	Owner        string
 	ProjectType  string
 	Stars        int
-	SDKVersion   int
+	SDKVersion   string
 	CreatedAt    string
 	PushedAt     string
 	TotalCommits int
 }
 
-//GetStats returns []RepoMetaData populated with Code results from Github Search Code API, for a given search string
-func GetStats(client *github.Client, q string) ([]RepoMetadata, error) {
-	var repoList []RepoMetadata
-	var projectType string
-	var sdkVersion int
-
-	//Search function return CodeResults for the q queery, and also Basic repository data.
-	searchOp, err := Search(client, q)
-
-	if err != nil {
-		return nil, err
-	}
-
-	if strings.Contains(q, "helm") {
-		projectType = "helm"
-	}
-	if strings.Contains(q, "ansible") {
-		projectType = "ansible"
-	}
-	if strings.Contains(q, "go.mod") {
-		projectType = "go.mod"
-	}
-	if strings.Contains(q, "Gopkg.toml") {
-		projectType = "Gopkg.toml"
-	}
-	for j := 0; j < len(searchOp); j++ {
-		for i := 0; i < len(searchOp[j].CodeResults); i++ {
-			owner := searchOp[j].CodeResults[i].GetRepository().GetOwner().GetLogin()
-			name := searchOp[j].CodeResults[i].GetRepository().GetName()
-			fullName := searchOp[j].CodeResults[i].GetRepository().GetFullName()
-
-			//getVersion returns SDK version by doing Fragment search
-			if projectType == "Gopkg.toml" {
-				sdkVersion = 0
-			} else {
-				sdkVersion = getversion(searchOp[j].CodeResults[i], projectType)
-			}
+//RepoMetadataQuery struct to store Search queries, and Project Type
+type RepoMetadataQuery struct {
+	ProjectType   string
+	Queries       []string
+	VersionParser VersionParser
+}
 
-			//GetRepoDetails returns Repository specififc details
-			repoMap := GetRepoDetails(client, owner, name)
+//VersionParser Interface implements ParseVersion, to Parse Version number from a given Text Match result.
+type VersionParser interface {
+	ParseVersion(codeResults github.CodeResult, projectType string) (string, error)
+}
 
-			TotalCommits, err := strconv.Atoi(repoMap["TotalCommits"])
-			Stars, err := strconv.Atoi(repoMap["Stars"])
-			if err != nil {
-				return nil, err
-			}
+//GetStats returns []RepoMetaData populated with Code results from Github Search Code API, for a given search string
+func GetStats(client *github.Client, rq RepoMetadataQuery) ([]RepoMetadata, error) {
+	var repoList []RepoMetadata
+	var sdkVersion string
+	var skipVersion int
 
-			repoDetails := &RepoMetadata{
-				FullName:     fullName,
-				ProjectType:  projectType,
-				SDKVersion:   sdkVersion,
-				Stars:        Stars,
-				CreatedAt:    repoMap["CreatedAt"],
-				PushedAt:     repoMap["PushedAt"],
-				TotalCommits: TotalCommits,
+	for _, q := range rq.Queries {
+		searchOp, err := Search(client, q)
+		if err != nil {
+			return nil, err
+		}
+		if strings.Contains(q, "Gopkg.toml") {
+			skipVersion = 1
+		}
+		for j := 0; j < len(searchOp); j++ {
+			for i := 0; i < len(searchOp[j].CodeResults); i++ {
+				if skipVersion == 1 {
+					sdkVersion = "N/A"
```

Comment:
This is starting to come into shape as a general purpose library, so should we use `version` as the variable name?

Reply:
I dont think I understood this, Can you please elaborate?

Reply:
Sorry, I'm just saying, lets change the variable name from "sdkVersion" to "version" now that this library his become much more extensible and usable for totally different projects.

Reply:
Will do.